pkg/api/graph/graph.go

```diff
@@ -353,10 +353,34 @@ func (g Graph) addEdges(edges []graph.Edge, fn EdgeFunc) {
 // node should be included.
 type NodeFunc func(g Interface, n graph.Node) bool
 
+// NewNodeFunc returns a new NodeFunc accepting the provided kinds of nodes
+func NewNodeFunc(kinds ...string) NodeFunc {
+	return func(g Interface, n graph.Node) bool {
+		for _, kind := range kinds {
+			if kind == g.Kind(n) {
+				return true
+			}
+		}
+		return false
+	}
+}
+
 // EdgeFunc is passed a new graph, an edge in the current graph, and should mutate
 // the new graph as needed. If true is returned, the existing edge will be added to the graph.
 type EdgeFunc func(g Interface, head, tail graph.Node, edgeKinds util.StringSet) bool
 
+// NewEdgeFunc returns a new EdgeFunc accepting the provided kinds of edges
+func NewEdgeFunc(kinds ...string) EdgeFunc {
+	return func(g Interface, head, tail graph.Node, edgeKinds util.StringSet) bool {
+		for _, kind := range kinds {
```

Comment:
Make the `util.StringSet` of the allowedKinds above, then run an `allowedKinds.HasAny(edgeKinds.List())`.  Also, can you open an issue indicating that we need to fix either `EdgeFunc` or `Subgraph` to properly handle cases where part of an edge should propogate?